src/plugin/physics/src/resource/PhysicsManager.hpp

```diff
@@ -81,6 +85,95 @@ class PhysicsManager {
      */
     inline void SetCollisionSteps(int steps) { _collisionSteps = steps; }
 
+    /**
+     * @brief Get the contact listener, casted back as a ContactListenerImpl.
+     *
+     * @return std::shared_ptr<Utils::ContactListenerImpl>
+     *
+     * @note This is safe because the contact listener is always created as a ContactListenerImpl.
+     */
+    inline std::shared_ptr<Utils::ContactListenerImpl> GetContactListener()
+    {
+        if (_contactListener == nullptr)
+        {
+            return nullptr;
+        }
+
+        return std::dynamic_pointer_cast<Utils::ContactListenerImpl>(_contactListener);
+    }
+
+    /**
+     * @brief Add a contact added callback to the contact listener.
+     *
+     * @param callback The callback to add.
+     *
+     * @return void
+     */
+    inline void AddContactAddedCallback(Utils::ContactListenerImpl::OnContactAddedCallback callback)
+    {
+        auto contactListener = GetContactListener();
+
+        if (contactListener != nullptr)
+        {
+            contactListener->AddOnContactAddedCallback(std::move(callback));
+        }
```

Comment:
Same a `GetContactListener` request. We maybe should do smth like:
```suggestion
#ifdef ES_DEBUG
        if (contactListener == nullptr)
        {
            ES::Utils::Log::Error(
                "PhysicsManager: tried to add contact added callback, but contact listener is not initialized.");
        }
#endif
        contactListener->AddOnContactAddedCallback(std::move(callback));
```

/!\ Do not accept this suggest, it will work but it will also duplicate code /!\

Reply:
nope, we want safety in release
with your request if the contactListener is called somehow by being null it will crash
right now it works a bit like OpenGL logic where it just fails silently in release and provides debug when asked for